フロントビュー実装のテスト

beginerbeginer opened the pull request:
#what
フロントビューの実装
ページを目コピして
html.haml
.scssのを書く
#why
フロントビューの確認をしてもらうため
hamlの練習のため
.scssの練習のため

beginerbeginer committed:
Merge branch 'master' into test

beginerbeginer committed:
Merge branch 'master' into test

beginerbeginer committed:
edit application.css

application.css 内の記述を全消去

beginerbeginer committed:
change application.css name

application.css
→
application.scss

beginerbeginer committed:
edit application.scss

add @import "./reset";

beginerbeginer committed:
add _reset.scss

さらに、リセットCSSをアプリケーションに反映させます。
まずはリセットCSSの記述を行うファイルを準備しましょう。
assets/stylesheetsディレクトリの中に_reset.scssというファイルを作成してください。
ChatSpaceでは、YUI 3のリセットファイルを使用します。以下のYUI3を先程作成した_reset.scssにコピーしてください。

https://master.tech-camp.in/curriculums/1154

beginerbeginer committed:
edit Gemfile

【日報】190211 13:59:33

  gem 'font-awesome-rails'  加えた

beginerbeginer committed:
add _variable.scss

https://master.tech-camp.in/curriculums/1154
【日報】190211 15:32:41

_variable.scss を作った

beginerbeginer committed:
add model message

【日報】190211 15:52:33
rake g model message をした

ビューのマークアップをここに書こうと思ったから

beginerbeginer committed:
edit index.html.haml

【日報】190211 19:08:28

テスト用のhtmlをかいた。

beginerbeginer committed:
add messages.scss

【日報】190211 19:09:06

テスト用CSS

beginerbeginer committed:
edit messages_controller.rb

【日報】190211 19:10:25
at terminal
rails g controller messages

beginerbeginer committed:
delete "message" controller

【日報】190211 19:11:12
s が足りなかったので
rails d controller message をした

beginerbeginer committed:
edit index.html.haml

【日報】190211 19:12:23
index.html.haml を追加
テスト用　のhtml.haml
をかいた

beginerbeginer committed:
edit index.html.haml

Haml::SyntaxError in Messages#index
Showing
/Users/masashimiyagaki/projects/chat-space2/app/views/messages/index.htm
l.haml where line #44 raised:

Self-closing tags can't have content.
Extracted source (around line #44):

beginerbeginer committed:
edit messages.scss

【日報】190211 20:07:19
sass がうまく html に反映されない

beginerbeginer committed:
edit application.scss

@import "./messages";
@import "./_variable";
@import "font-awesome";

加えた
まだ書く意味がわからない
→質問して聞いてみる。

beginerbeginer committed:
edit config/application.rb

  g.stylesheets false
 が多かったので消した。

→メンタに質問するつもり

beginerbeginer committed:
edit app/assets/stylesheets/messages.scss

}
が少なかったので書き足した。
とりあえずCSSの反映はできた

beginerbeginer committed:
edit config/application.rb

      g.stylesheets false
特に影響がない
→戻した。

beginerbeginer commented:
フロントビューのコード
反映させるまでは書きました。
中途半端ですみません

yuki-uchida commented:
Gyazoなどで実際に作っていただいたビューのデザインが崩れていないかを見せていただきたいです！

yuki-uchida commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,128 @@
+  li {
+    text-align: -webkit-match-parent;
+  }
+  // .header_icons {
```
コメントアウトは消しましょう

yuki-uchida commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,128 @@
+    height:calc(100vh-100px);
+    padding:0 20px;
+  }
+  .group_name {
```
全体的に、https://qiita.com/Takuan_Oishii/items/0f0d2c5dc33a9b2d9cb1 に沿ってクラスの命名を修正しましょう

yuki-uchida commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,128 @@
+
+    .edit__submit {
+      color: #38AEF0;
+      background: #FFFFFF;
```
色は変数化しておきましょう。色管理用のscssを作成しておくことによってサイト全体の色統一、変更が容易になり可読性も上がります

beginerbeginer committed:
edit index.html.haml

【日報】190212 13:51:26

書き直した。
jsfiddleを使いながらかいている
何を直せばいいか把握できていない。
とりあえず今の画面をgazzoで送る

beginerbeginer committed:
edit messages.scss

【日報】190212 13:54:49
コメントアウトを消した。

beginerbeginer commented:
![Uploading c53d185b0b633b7da019430ee3eb4604.png…]()

beginerbeginer commented:
https://gyazo.com/c53d185b0b633b7da019430ee3eb4604
課題点
・ebm がまだわかってない
・_color_variable.scss
のメリットがわかるが
どうやって
何を書けばいいかわからない

yusukeyamane commented:
見せて頂いたスクリーンショットだとかなり見本と差異があるように見受けられます！
完成した後改めてPRを出して頂いても良いでしょうか？？

beginerbeginer committed:
edit index.html.haml

変換を使った
まだ理解できていないが
EBM が大切な概念であることはわかつた。

beginerbeginer committed:
edit _variable.scss

variable.scss
→
_color_variable.scss

beginerbeginer committed:
edit application.scss

application.scss
を編集した。

beginerbeginer committed:
edit messages.scss

beginerbeginer commented:
[![Screenshot from Gyazo](https://gyazo.com/727861577d6520217e447aa7b4473eb0/raw)](https://gyazo.com/727861577d6520217e447aa7b4473eb0)

beginerbeginer committed:
edit messages.scss

コメントアウトを消した。

beginerbeginer commented:
cssの調整わからないので何か参考になるサイトなどはありますか？
コメントアウトは消しました。
よろしくお願いいたします。

yusukeyamane commented:
・カリキュラムのhtml, css部分を見直していただくこと
・http://ysd-lab.jp/markup/learn.html こういった実践形式のサイトを取り組んで見ること
・残りわからない部分はhttp://www.htmq.com/csskihon/001.shtml から該当箇所を検索して実装していく

という方針になるかと思います！もしもっと具体的に方針を決めたいということであれば担当のライフコーチにご相談ください！（彼らの方がより細かいサポートができると思います！）

beginerbeginer committed:
edit index.html.haml

【日報】190212 18:40:54習字のようにまずは完成形を見て
なぜそうなるのかを考えて
もう一度自分で書くほうがいいと思い
そのまま拝借したものをhaml で書いた

beginerbeginer committed:
edit messages.scss

messages.scssを全て消して
hamlのみでどう反映されるかを見てみた。
sassの書き方にはまだできない

yusukeyamane commented on app/assets/stylesheets/_color_variable.scss:
```diff
@@ -0,0 +1,20 @@
+$black-flash: #434a54;
+$alert_orange: #F35500;
+$button-white: white;
+
```
whiteと#fffとFFFFFFは一緒の色なので一つの変数でいいですね。
他にも同じ意味のものは同じ変数でいいです！

beginerbeginer replied:
ありがとうございます。
アドバイスあると助かります。

beginerbeginer committed:
edit messages.scss

css を加えた。

beginerbeginer committed:
edit messages.scss

【日報】190212 18:56:58

yamane さんからのアドバイス
同じいみ色はまとめられる
→そこから得られる教訓

同じことはまとめた方が効率的。

beginerbeginer commented:
![2019-02-12 18 42 29](https://user-images.githubusercontent.com/43893159/52627219-3f489800-2ef8-11e9-8e0d-3a1889943772.png)

beginerbeginer commented:
https://i.gyazo.com/58e094798ab5ab795e52e52051b6e49e.png

beginerbeginer commented:
![2019-02-12 18 42 29](https://user-images.githubusercontent.com/43893159/52627410-a1090200-2ef8-11e9-8dcd-ac245ad6187b.png)

習字みたいに何度か繰り返すしかないと思いました。
目コピで作成は自分ではまだ厳しい。
要素をどう分解するべきかの練習が足りない。
わかることと理解の差が大きいことを理解した。
何度か繰り返す必要があるため今回はほぼ真似て作った

beginerbeginer commented:
![2019-02-12 19 04 44](https://user-images.githubusercontent.com/43893159/52627587-1674d280-2ef9-11e9-8e8c-1b7b6b2e84a7.png)

beginerbeginer committed:
edit messages.scss →messages.css

【日報】190212 20:25:44
自分で実際に(現時点)
EBMでかく現実的な方法は

コンソールにあるCSSを書き換え亭いく方法がある意味手っ取り早そうだ。

beginerbeginer committed:
delete messages.scss

当たり前のことだが
scss →cssに変えて
コピペして貼り付けた
どうしてもわからなかった。

beginerbeginer commented:
すいません。あまりにも本番環境のCSSをコンソールで調べて
EBM に従って書き換える方法は現実的でしょうか？

感想
![2019-02-12 20 33 50](https://user-images.githubusercontent.com/43893159/52632691-930dae00-2f05-11e9-8416-ffc584f13b34.png)

習字みたいに何度か繰り返すしかないと思いました。
目コピに必要な能力
→ページをボックスで分割する能力
→そのボックスにどんなCSSが付いているか気づく能力
わからない原因
要素をどう分解するべきかの練習が足りない。
わかることと理解の差が大きい
何度か繰り返す必要があるため今回はほぼ真似て作った。

ReoKuriyama commented:
経験が少ない内は真似た方が近道になると思います。
`div`を使ってどうページを分けていくかなどは実際の使われ方をみないと難しいところがあるので。
ただし、コードを写すときは何を意味しているかを把握しながらにしましょう

ReoKuriyama commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,49 @@
+            .main-header__edit-btn Edit
+        .messages
+        .form
+          %form#new_message.new_message{"accept-charset" => "UTF-8", :action => "/groups/149/messages", :enctype => "multipart/form-data", :method => "post"}
```
以下三行は手動で書くものではないです
これらはform_tag等で生成されるものです。
写すのは最悪良いですが何を意味しているか分からないまま書いても身になりません

beginerbeginer committed:
delete app/assets/stylesheets/_color_variable.scss

beginerbeginer committed:
add edit app/assets/stylesheets/_reset.scss

beginerbeginer committed:
edit app/assets/stylesheets/application.scss

CSSをそれぞれの役割で分けることが大切。
色
文字の大きさ
reset.css
clearfix
引数と値
サイドバの部品
メッセージをの部品
メッセージ用のCSS

に分けていた。
答え見ながらやりました。
２回目は自分で書き直します。

beginerbeginer committed:
edit app/assets/stylesheets/config/colors.scss

色を引数を作り定義した
場所_色
とかくとわかりやすいようだ。

beginerbeginer committed:
edit font-size.scss

色と同様に
場所_文字大きさ
場所_名前
場所_グループメッセージ
場所と　文字の属性がわかる
→イメージしやすく書く

beginerbeginer committed:
delete app/assets/stylesheets/messages.css

beginerbeginer committed:
edit app/assets/stylesheets/messages/messages.scss

なぜそうしたのかよくわかってない場所

beginerbeginer committed:
edit mixins/mixin.scss

add edit /mixins/mixin.scss
@mixin @include
の使い方は重要

beginerbeginer committed:
edit mixins/vendor_prefix.scss

引数の意図がわからない
$prefixes: -webkit-, -moz-, -ms-, -o-, '';

scss とここでの  @each の関係は？
https://www.nxworld.net/tips/sass-each-and-for-snippets.html
Sassには@eachや@forといった制御構文があり、それらを利用することでCSSで繰り返し記述していたような手間を省くことができます。

beginerbeginer committed:
edit  modules/chat.scss

BEMを意識した書き方
BEM(BlockとElementをつなぐ場合は、アンダースコア2つでつなぐ）
ルールが把握できてきた。

beginerbeginer committed:
edit modules/side_bar.scss

サイドバのcss
にモジュール化して書くほうがわかりやすいことに気づけた。
今回は左のサイドバには
グループがあるのそこにかかるcssも書いた。

beginerbeginer committed:
edit application.html.haml

    = stylesheet_link_tag    'application', media: 'all',
'data-turbolinks-track' => true
    = javascript_include_tag 'application', 'data-turbolinks-track' =>
true

reload と　true がどう違うのかがわからない。

beginerbeginer committed:
edit mailer.html.haml

特になし

beginerbeginer committed:
edit app/views/messages/index.html.haml

!!!
がなくてもよかったことに気づけた問題なくページは見える。
あくまでhtmlで書いてあることを意味するだけだから関係ないのかも。

.wrapperがなぜ必要なのかがまだわからない。

beginerbeginer committed:
edit views/shared/_side_bar.html.haml

部分テンプレートとを置く場所が一般的に
app/views/shared/_~~~

と書くことを知った。
今回はサイドバーのHTML

ここは繰り返しが使える部分
次は自分でかけるように練習する

beginerbeginer commented:
![2019-02-13 18 02 36](https://user-images.githubusercontent.com/43893159/52699767-dde8fd80-2fb9-11e9-90b0-2a775564c614.png)

cssの機能ごとに分けて書くとわかりやすくなることに気づけた。
htmlは場所→そこにあるクラスについて書くと場所が視覚的にイメージしやすいことに気づけた。
グループの場所は部分テンプレートが使える。
wrapperを使った理由がまだわからない。

TOMARUMARU commented on app/views/layouts/mailer.html.haml:
```diff
@@ -1,8 +1,8 @@
 !!!
```
mailerファイルは使用しないので消してしまいましょう

beginerbeginer replied:
修正しました。確認よろしくお願い致します。

TOMARUMARU commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,62 @@
+
```
改行が空きすぎですね。
改行やインデントもきちんと揃えましょう

beginerbeginer replied:
修正しました。確認よろしくお願い致します。

beginerbeginer committed:
delete  mailer.html.haml/mailer.text.haml

mailerファイルは使用しないので消去した。

beginerbeginer committed:
edit index.html.haml

余計な改行を消した。

beginerbeginer committed:
edit index.html.haml

改行をした

beginerbeginer committed:
edit index.html.haml

beginerbeginer commented:
修正しました。確認よろしくお願い致します。

TOMARUMARU reviewed (approved):
LGTMです！
